api.py

```diff
@@ -0,0 +1,32 @@
+import json
+from django.contrib.auth.decorators import user_passes_test
+from django.core import serializers
+from django.db import transaction, connection
+from django.http import HttpResponse
+from django.views.decorators.cache import never_cache
+from django.views.decorators.csrf import csrf_protect
+
+import commands
+
+@csrf_protect
+@never_cache
+@transaction.atomic
+@user_passes_test(lambda u: u.is_staff)
+def command(request):
+	data = json.loads(request.POST.get('data', '{}'))
+	func = getattr(commands, data['command'], None)
+	status = 200
+	if func:
+		if request.user.has_perm(func.permission):
+			output, status = func(data)

```

Comment:
Somewhat related to my previous comment, I feel we should wrap the command invocation with a universal try/except block, and have it return a standardized error message as a json return (while still logging it somewhere) to help with debugging? Maybe something to consider for the future, since I suspect you want to get this stuff out by yesterday at this point.